utils/smartctl.go

```diff
@@ -84,24 +86,24 @@ func (s *Smartctl) Attributes() (utilName model.CollectorUtility, absolutePath s
 }
 
 // Drives returns drives identified by smartctl
-func (s *Smartctl) Drives(ctx context.Context) ([]*common.Drive, error) {
-	drives := make([]*common.Drive, 0)
-
+func (s *Smartctl) Drives(ctx context.Context) ([]*drive.Drive, error) {
 	DrivesList, err := s.Scan(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, drive := range DrivesList.Drives {
+	drives := make([]*drive.Drive, 0, len(DrivesList.Drives))
+	for _, d := range DrivesList.Drives {
 		// collect drive information with smartctl -a <drive>
-		smartctlAll, err := s.All(ctx, drive.Name)
+		smartctlAll, err := s.All(ctx, d.Name)
 		if err != nil {
 			return nil, err
 		}
 
 		item := &common.Drive{
 			Common: common.Common{
-				Vendor:      common.VendorFromString(smartctlAll.ModelName),
+				LogicalName: d.Name,
```

Comment:
This is necessary to be able to wipe disks w/o having to store the logicalname externally (a map maybe). Its also necessary because this is deep in ironlib and we lose the logicalname once we return to the caller. It seems to me that the lower layers should have all the data and higher level should be in charge or removing things it doesn't want. It definitely makes sense to *not* send LogicalName to an external store but that should be take care of in the sending side not data collection side right?